src/main/java/com/minecolonies/coremod/commands/CommandEntryPointNew.java

```diff
@@ -1,573 +1,86 @@
-    {
-        final Map<String, ActionMenuHolder> map = new HashMap<>();
-        final IMenu menu = treeNode.getData();
-        if (!menu.getMenuType().isNavigationMenu())
+        @Override
+        public boolean canUseCommands(MinecraftServer server, ICommandSender sender)
         {
-            final ActionMenu actionMenu = (ActionMenu) menu;
-            for (final ActionArgument actionArgument : actionMenu.getActionArgumentList())
+            if (sender instanceof EntityPlayer)
             {
-                if (usedThisActionArgumentName(parsedHolders, actionArgument))
-                {
-                    continue;
-                }
-                map.put(actionArgument.getName().toLowerCase(Locale.ROOT) + ":", new ActionMenuHolder(treeNode, actionArgument));
+                return AbstractSingleCommand.isPlayerOpped(sender) || Configurations.gameplay.opLevelForServer <= 0;
             }
+            return true;
         }
-        return map;
     }
 
-    private boolean usedThisActionArgumentName(@NotNull final List<ActionMenuHolder> parsedHolders, @NotNull final ActionArgument actionArgument)
+    protected ForgeCommandsPermissionsChecker getPermissionsChecker()
     {
-        for (final ActionMenuHolder actionMenuHolder : parsedHolders)
-        {
-            if (actionMenuHolder.getActionArgument().getName().equals(actionArgument.getName()))
-            {
-                return true;
-            }
-        }
-        return false;
+        return new ForgeCommandsPermissionsChecker();
     }
 
-    @NotNull
-    @Override
-    public List<String> getTabCompletions(
-                                           @NotNull final MinecraftServer server,
-                                           @NotNull final ICommandSender sender,
-                                           @NotNull final String[] args,
-                                           @Nullable final BlockPos pos)
+    protected NavigationMenuType getRootNavigationMenuType()
     {
-        final ParsingResult parsingResult = getTabCompletionsAndParsingHolders(root, server, sender, args, pos);
-        return parsingResult.getTabCompletions();
+        return NavigationMenuType.MINECOLONIES;
     }
-
-    @NotNull
-    private ParsingResult getTabCompletionsAndParsingHolders(
-                                           @NotNull final TreeNode<IMenu> treeNode,
-                                           @NotNull final MinecraftServer server,
-                                           @NotNull final ICommandSender sender,
-                                           @NotNull final String[] args,
-                                           @Nullable final BlockPos pos)
+    
+    protected ModuleContext getModuleContext()
     {
-        if (treeNode.getData().getMenuType().isNavigationMenu())
+        return new ModuleContext()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Make this anonymous inner class a lambda (sonar.java.source not set. Assuming 8 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1604)